### PR TITLE
fix overflow in array length computation

### PR DESCRIPTION
**Pull request by reez12g**
addressing https://github.com/rust-lang/rust/issues/114912

### CODE REVIEW

File: compiler/rustc_hir_typeck/src/generator_interior/mod.rs [hunk 1]
```diff
@@ -643,15 +643,19 @@ fn check_must_not_suspend_ty<'tcx>(
         }
         ty::Array(ty, len) => {
             let descr_pre = &format!("{}array{} of ", data.descr_pre, plural_suffix);
+            let target_usize = len.try_eval_target_usize(fcx.tcx, fcx.param_env).unwrap_or(0);
+            let plural_len = if target_usize == usize::MAX {
+                usize::MAX
+            } else {
+                computed_len + 1
+            };
```
Comment (PatchMixolydic) [inline compiler/rustc_hir_typeck/src/generator_interior/mod.rs:651]: This is essentially [`usize::saturating_add`](https://doc.rust-lang.org/std/primitive.usize.html#method.saturating_add):
```suggestion
            let target_usize = len.try_eval_target_usize(fcx.tcx, fcx.param_env).unwrap_or(0) as usize;
            let plural_len = target_usize.saturating_add(1);
```

File: compiler/rustc_hir_typeck/src/generator_interior/mod.rs [hunk 2]
```diff
@@ -643,15 +643,19 @@ fn check_must_not_suspend_ty<'tcx>(
...
             check_must_not_suspend_ty(
                 fcx,
                 ty,
                 hir_id,
                 SuspendCheckData {
                     descr_pre,
-                    plural_len: len.try_eval_target_usize(fcx.tcx, fcx.param_env).unwrap_or(0)
-                        as usize
-                        + 1,
+                    plural_len: plural_len,
```
Comment (PatchMixolydic) [inline compiler/rustc_hir_typeck/src/generator_interior/mod.rs:658]:
```suggestion
                    plural_len,
```

**Review by davidtwco — APPROVED** [review]

**Comment by davidtwco** [comment]
Looks like you'll need to run `./x.py fmt`.

**Comment by reez12g** [comment]
@davidtwco 
Thanks!
I run `./x.py fmt`

**Review by davidtwco — APPROVED** [review]

**Comment by compiler-errors** [comment]
why was there not a test added in this PR?

**Pull request merged**